src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/CodeModelToAST.java

```diff
@@ -45,67 +42,26 @@ public CodeModelToAST(TreeMaker treeMaker, Names names, Symtab syms, Resolve res
-                ExternalizableOp.class, RecordTypeRef.class
-        };
-        for (Class<?> crType : crTypes) {
-            JavaType jt = JavaType.type(crType.describeConstable().get());
-            Type t = syms.enterClass(jdk_incubator_code, jt.externalize().toString());
-            m.put(jt, t);
-        }
-        Class<?>[] javaBaseTypes = {
-                HashMap.class, String.class, Object.class, Map.class, java.util.List.class
-        };
-        for (Class<?> javaBaseType : javaBaseTypes) {
-            JavaType jt = JavaType.type(javaBaseType.describeConstable().get());
-            Type t = syms.enterClass(syms.java_base, jt.externalize().toString());
-            m.put(jt, types.erasure(t));
-        }
-
-        m.putAll(Map.ofEntries(
-                Map.entry(JavaType.BOOLEAN, syms.booleanType),
-                Map.entry(JavaType.BYTE, syms.byteType),
-                Map.entry(JavaType.SHORT, syms.shortType),
-                Map.entry(JavaType.CHAR, syms.charType),
-                Map.entry(JavaType.INT, syms.intType),
-                Map.entry(JavaType.LONG, syms.longType),
-                Map.entry(JavaType.FLOAT, syms.floatType),
-                Map.entry(JavaType.DOUBLE, syms.doubleType)
-        ));
-
-        return m;
     }
 
     private Type typeElementToType(TypeElement te) {
         JavaType jt = (JavaType) te;
         return switch (jt) {
+            case PrimitiveType pt when pt == JavaType.BOOLEAN -> syms.booleanType;
+            case PrimitiveType pt when pt == JavaType.BYTE -> syms.byteType;
+            case PrimitiveType pt when pt == JavaType.CHAR -> syms.charType;
+            case PrimitiveType pt when pt == JavaType.INT -> syms.intType;
+            case PrimitiveType pt when pt == JavaType.LONG -> syms.longType;
+            case PrimitiveType pt when pt == JavaType.FLOAT -> syms.floatType;
+            case PrimitiveType pt when pt == JavaType.DOUBLE -> syms.doubleType;
             case ClassType ct when ct.hasTypeArguments() -> {
-                ClassType enclosingType = ct.enclosingType().orElse(null);
-                ListBuffer<Type> typeArgs = new ListBuffer<>();
-                for (JavaType typeArgument : ct.typeArguments()) {
-                    typeArgs.add(typeElementToType(typeArgument));
-                }
-                yield new Type.ClassType(typeElementToType(enclosingType), typeArgs.toList(),
-                        typeElementToType(ct.rawType()).tsym);
+                Type enclosing = ct.enclosingType().map(this::typeElementToType).orElse(Type.noType);
+                List<Type> typeArgs = List.from(ct.typeArguments()).map(this::typeElementToType);
+                yield new Type.ClassType(enclosing, typeArgs, typeElementToType(ct.rawType()).tsym);
             }
+            case ClassType ct -> types.erasure(syms.enterClass(attrEnv.toplevel.modle, ct.toClassName()));
```

Comment:
note: the module passed here `attrEnv.toplevel.modle` is *not* the module the type we are looking for belongs to. It is the module of the current compilation unit -- in other words, this module is used to determine which packages are visible from the compilation unit.